### ISSUE TITLE
unable to use CUVID (MPS-HC crashes)

### ISSUE THREAD

**Issue by typhoon71** [Nevcairiel/LAVFilters#48]
I just discovered that playing HEVC using CUVID decoder crashes MPC-HC.
I don't know if it's caused by CUVID, MPC or LAV (Video decoder), but I suppose I can start here.

In the past I did set it up to play 1080 and 4k using the GPU via CUVID, and it di play the test vids.
Now I used a HEVC encoded file and MPC-HC just crashes; I discovered I can play the file with CPU decoding, but not with the GPU (CUVID).

My GTX660 is supposed to be able to play 1920x1080 HEVC video so I'm puzzled. 
Here's a dump: https://drdump.com/UploadedReport.aspx?DumpID=5713660&SecondVisit=1

It seems to point to some CUVID thing, and being LAV Video decoder the only thing that uses it I suppose it's a bug? I hope it's something that's possible to fix...

(Windows 7 64bit, NV Drivers 359.00, MPC-HC 1.7.10.28 (0115ec4), LAV 0.67.0.3-git, madvr 0.89.17)
If you need other data just ask.

**Comment by Nevcairiel** [comment]
CUVID crashing in this manner is a known problem, and unfortunately a bug in NVIDIAs driver. NVIDIA has stopped developing CUVID further, and the more time goes by, the more issues show up in it.

I recommend to switch to DXVA2-CB instead, there is nothing I can do to fix CUVID, since its a NVIDIA component.

**Comment by typhoon71** [comment]
Tch, bummer. If I could I'd slap them (NVIDIA) gently with a iron bar.
I'll use DXVA2-CB as you suggested, and thanks for the tip.

**Comment by Nevcairiel** [comment]
You could try the latest nightly from https://files.1f0.de/lavf/nightly/ , I tried something, maybe it helps even if its unlikely.

**Comment by typhoon71** [comment]
I tried the new nightly but it still crashes.

I did also discover that my card does not support 10bit HEVC... and that video was indeed a 10bit HEVC.
Basically I noticed (with DXVA checker) that the card (GTX660) supports "HEVC_VLD_Main" but not "HEVC_VLD_Main10" mode. And I'm pretty sure those tests I made long ago didn't include 10bit HEVC.

So maybe a sensible solution would be to check for 10bit and act in a conservative manner if it's not supported by the card (DXVA2-CB just uses avcodec for that video, and it's the reason I found out about the 10bit matter)
